src/drivers/actuators/vertiq_io/vertiq_io.cpp

```diff
@@ -0,0 +1,306 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2012-2022 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+#include "vertiq_io.hpp"
+
+#include <px4_platform_common/log.h>
+
+px4::atomic_bool VertiqIo::_request_telemetry_init{false};
+char VertiqIo::_telemetry_device[] {};
+
+VertiqIo::VertiqIo() :
+	OutputModuleInterface(MODULE_NAME, px4::wq_configurations::hp_default),
+	_serial_interface(0),
+	_client_manager(&_serial_interface),
+	_telem_manager(_client_manager.GetMotorInterface())
+
+{
+	//Make sure we get the correct initial values for our parameters
+	updateParams();
+
+	_client_manager.Init((uint8_t)_param_vertiq_target_module_id.get());
+	_motor_interface_ptr = _client_manager.GetMotorInterface();
+
+	_serial_interface.SetNumberOfClients(_client_manager.GetNumberOfClients());
```

Comment:
we've figured out how many clients we're dealing with. tell the serial interface.